src/simsopt/field/magneticfieldclasses.py

```diff
@@ -862,10 +862,14 @@ def __init__(self, field, degree, rrange, phirange, zrange, extrapolate=True, nf
         Args:
             field: the underlying :mod:`simsopt.field.magneticfield.MagneticField` to be interpolated.
             degree: the degree of the piecewise polynomial interpolant.
-            rrange: a 3-tuple of the form ``(rmin, rmax, nr)``. This mean that the interval :math:`[rmin, rmax]` is
-                    split into ``nr`` many subintervals.
-            phirange: a 3-tuple of the form ``(phimin, phimax, nphi)``.
-            zrange: a 3-tuple of the form ``(zmin, zmax, nz)``.
+            rrange: a 3-tuple of the form ``(rmin, rmax, nr)`` or a 4-tuple of the form ``(rmin, rmax, nr, include_endpoint)``.
+            The default for `include_endpoint` is True. This mean that the interval :math:`[rmin, rmax]` is
+            split into ``nr`` many subintervals. When `include_endpoint` is False, interval :math:`[rmin, rmax)` is
+            split into ``nr`` many subintervals.
```

Comment:
what does it mean [rmin, rmax]` is split into ``nr`` many subintervals as opposed to `[rmin, rmax)` is split into ``nr`` many subintervals ?

Reply:
To concretize this, lets suppose you want 5 points between 0 and 1. If you include the end point, the mesh would [0, 0.25, 0.5, 0.75, 1]. If you don't want to include the end point, the mesh generated would be [0, 0.2, 0.4, 0.6, 0.8]. So the mesh would be quite different. Unfortunately, BMW generates phi grid with end point not included.

Reply:
But then what happens on the small subinterval (2pi/nfp - h, 2pi/nfp)?  I guess the behavior would be dictated by the `extrapolate` keyword argument.

Since the BMW field is nfp-periodic, you can just rotate the B-field values from B(R, phi=0, Z) to B(R, phi=2pi/nfp, Z) and obtain the endpoint value.